internal/e2e/profiler/profiler.go

```diff
@@ -58,8 +58,10 @@ func main() {
 	p := wzprof.ProfilingFor(guestBin)
 	cpu := p.CPUProfiler()
 	mem := p.MemoryProfiler()
-	ctx := context.WithValue(context.Background(),
-		experimental.FunctionListenerFactoryKey{},
+
+	// Use `WithFunctionListenerFactory` instead of ``FunctionListenerFactoryKey`.
```

Comment:
You may not need this comment because this is very minor change.

Reply:
@Gekko0114 
I will remove this, thanks